sched/validator.cpp

```diff
@@ -177,10 +182,52 @@ static inline void is_invalid(DB_HOST_APP_VERSION& hav) {
     }
 }
 
+// check for results with long-term failure; punish those hosts
+//
+void scan_punitive(vector<VALIDATOR_ITEM>& items) {
+    void* data=NULL;
+    char buf[256];
+
+    for (unsigned int i=0; i<items.size(); i++) {
+        RESULT& result = items[i].res;
+        if (result.server_state != RESULT_SERVER_STATE_OVER) continue;
+        if (result.outcome != RESULT_OUTCOME_CLIENT_ERROR) continue;
+        if (result.validate_state != VALIDATE_STATE_INIT) continue;
```

Comment:
The failing tasks I am looking at have the state VALIDATE_STATE_INVALID before the validator is run rather than VALIDATE_STATE_INIT